### PR TITLE
Prevent VS2017 from emitting ambiguous symbol errors

### PR DESCRIPTION
**Pull request by sgolodetz**
These `std::forward` calls cause VS2017 to emit:

    error C2872: 'std': ambiguous symbol

This fix prevents the ambiguity by specifying that `::std` is intended.

### CODE REVIEW

**Comment by ezyang** [comment]
Blegh. Under what situations does this trigger?

**Comment by ezyang** [comment]
I am not in principle opposed to this patch, but there needs to be a comment explaining why the qualification is necessary

**Comment by sgolodetz** [comment]
It triggers for me even with this minimal compilable example:

    #include <torch/torch.h>

    int main()
    {
      return 0;
    }

In case it's relevant, I'm using VS2017 version 15.7.3.

**Comment by sgolodetz** [comment]
The full log is:

> 1>------ Build started: Project: scratchtest_torch, Configuration: Release x64 ------
> 1>main.cpp
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch\csrc\api\include\torch/nn/pimpl.h(123): error C2872: 'std': ambiguous symbol
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch/csrc/jit/argument_spec.h(380): note: could be 'std'
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch\csrc\api\include\torch/nn/pimpl.h(123): note: or       'std'
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch\csrc\api\include\torch/nn/modules/batchnorm.h(92): note: see reference to class template instantiation 'torch::nn::ModuleHolder<torch::nn::BatchNormImpl>' being compiled
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch/csrc/jit/stack.h(30): note: see reference to class template instantiation 'c10::ArrayRef<c10::IValue>' being compiled
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch\csrc\api\include\torch/nn/pimpl.h(129): error C2872: 'std': ambiguous symbol
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch/csrc/jit/argument_spec.h(380): note: could be 'std'
> 1>C:\Users\Stuart Golodetz\Downloads\libtorch\include\torch\csrc\api\include\torch/nn/pimpl.h(129): note: or       'std'
> 1>Done building project "scratchtest_torch.vcxproj" -- FAILED.
> ========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

**Comment by sgolodetz** [comment]
I get the same error if I include `<utility>` before `torch/torch.h`.

**Review by ezyang — APPROVED** [review]
Blast if I know why. Approved. (you'll need to get someone else to merge though)

**Comment by sgolodetz** [comment]
Thanks. I wasn't sure why either to be honest - but this fixes the problem, and should be entirely safe. The comments should allow anyone who comes up with a better fix to remove it again later without worrying if need be.

**Comment by goldsborough** [comment]
LGTM. Also reported in https://discuss.pytorch.org/t/there-is-an-error-c2872-in-pimpl-h-line-123-and-line129/33630. Windows is weird.

**Review by goldsborough — APPROVED** [review]

**Comment by iwroldtan** [comment]
> These `std::forward` calls cause VS2017 to emit:
> 
> ```
> error C2872: 'std': ambiguous symbol
> ```
> This fix prevents the ambiguity by specifying that `::std` is intended.

It can be solved by specify ‘std’ of the vs, example '::std'.
